core/app/models/mno_enterprise/user.rb

```diff
@@ -279,5 +284,44 @@ def staff?
     def admin?
       admin_role.to_s.casecmp(ADMIN_ROLE).zero?
     end
+
+    def requires_otp_for_login?
+      if admin?
+        Settings.authentication.two_factor.admin_enabled
+      else
+        Settings.authentication.two_factor.users_enabled
+      end
+    end
+
+    def activate_otp
+      self.attributes = {
+        otp_required_for_login: true,
+        unconfirmed_otp_secret: nil
+      }
```

Comment:
Do we want to just set this here? Or better to merge? Not sure if there will be other attributes that are needed (now or in the future) might be better to merge just in case.